request.go

```diff
@@ -1,10 +1,14 @@
 package coap
 
-import "context"
+import (
+	"context"
+	"crypto/x509"
+)
 
 type Request struct {
-	Msg      Message
-	Client   *ClientConn
-	Ctx      context.Context
-	Sequence uint64 // discontinuously growing number for every request from connection starts from 0
+	Msg              Message
+	PeerCertificates []*x509.Certificate
```

Comment:
I think PeerCertificates doesn't bellows to Request.  It will be better as receiver function in ClientConn (just forward calls via commenader to the session, and then to the connection.